src/gluonnlp/models/t5.py

```diff
@@ -0,0 +1,849 @@
+            flatten=False, 
+            use_bias=False, 
+            weight_initializer=Normal(self._inner_dim ** -0.5 * init_factor), 
+            dtype=dtype
+        )
+        if is_decoder: 
+            self.cross_attn_layer_norm = T5LayerNorm(
+                d_model=d_model, 
+                epsilon=layer_norm_eps, 
+                init_factor=init_factor, 
+                dtype=dtype
+            )
+            # avoid scaling before softmax
+            self.cross_attn_q = nn.Dense(
+                units=self._inner_dim, 
+                in_units=d_model, 
+                flatten=False, 
+                use_bias=False, 
+                weight_initializer=Normal((d_model * d_kv) ** -0.5 * init_factor), 
+                dtype=dtype
+            )
+            self.cross_attn_k = nn.Dense(
+                units=self._inner_dim, 
+                in_units=d_model, 
+                flatten=False, 
+                use_bias=False, 
+                weight_initializer=Normal(d_model ** -0.5 * init_factor), 
+                dtype=dtype
+            )
+            self.cross_attn_v = nn.Dense(
+                units=self._inner_dim, 
+                in_units=d_model, 
+                flatten=False, 
+                use_bias=False, 
+                weight_initializer=Normal(d_model ** -0.5 * init_factor), 
+                dtype=dtype
+            )
+            self.cross_attn = MultiHeadAttentionCell(
+                query_units=self._inner_dim, 
+                num_heads=num_heads, 
+                attention_dropout=dropout_prob, 
+                scaled=False, 
+                normalized=False, 
+                dtype=dtype, 
+                layout='NTK' if layout == 'NT' else 'TNK', 
+                use_einsum=False
+            )
+            self.cross_attn_proj = nn.Dense(
+                units=d_model, 
+                in_units=self._inner_dim, 
+                flatten=False, 
+                use_bias=False, 
+                weight_initializer=Normal(self._inner_dim ** -0.5 * init_factor), 
+                dtype=dtype
+            )
+        self.ffn = T5FeedForward(
+            d_model=d_model, 
+            d_ff=d_ff, 
+            dropout_prob=dropout_prob, 
+            layer_norm_eps=layer_norm_eps, 
+            activation=activation, 
+            init_factor=init_factor, 
+            dtype=dtype
+        )
+        self.dropout = nn.Dropout(dropout_prob)
+
+    @property
+    def layout(self): 
+        return self._layout
+
+    @_assert_decoder_method
+    @property
+    def state_batch_axis(self): 
+        if self.layout == 'NT': 
+            return 0, 0
+        else: 
+            return 1, 1
+
+    @_assert_decoder_method
+    def _init_states(self, batch_size, ctx, dtype='float32'): 
+        if self.layout == 'NT': 
+            shape = (batch_size, 0, self._num_heads, self._d_kv)
+        else: 
+            shape = ((0, batch_size, self._num_heads, self._d_kv))
```

Comment:
Need to drop the additional tuple:

`((0, batch_size, self._num_heads, self._d_kv))` --> `(0, batch_size, self._num_heads, self._d_kv)`